File: tests/e2e/auth_flow.mjs
import { chromium } from 'playwright';

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();
  
  console.log('--- Iniciando Teste E2E: Redefinição de Senha ---');
  
  try {
    // 1. Acessar página de login admin
    await page.goto('http://localhost:8080/admin-login');
    console.log('Acessou /admin-login');

    // 2. Clicar em esqueci minha senha
    await page.click('button:has-text("Esqueci minha senha admin")');
    console.log('Clicou em "Esqueci minha senha"');

    // 3. Tentar usuário inexistente
    await page.fill('input[placeholder="Seu usuário admin"]', 'usuario_errado');
    await page.click('button:has-text("Verificar usuário")');
    // Esperar mensagem de erro (no código atual deve falhar se não for 'admin')
    const errorMsg = await page.textContent('.alert-error');
    console.log('Erro usuário inválido:', errorMsg);

    // 4. Fluxo correto
    await page.fill('input[placeholder="Seu usuário admin"]', 'admin');
    await page.click('button:has-text("Verificar usuário")');
    console.log('Usuário "admin" verificado');

    // 5. Definir nova senha
    await page.fill('input[placeholder="Nova senha personalizada"]', 'nova_senha_2026');
    await page.click('button:has-text("Salvar Nova Senha")');
    console.log('Senha alterada com sucesso');

    // 6. Tentar logar com a nova senha
    await page.click('button:has-text("Voltar para o login")');
    await page.fill('input[placeholder="Usuário"]', 'admin');
    await page.fill('input[placeholder="Senha"]', 'nova_senha_2026');
    await page.click('button[type="submit"]');
    
    // Verificar se entrou no /admin
    await page.waitForURL('**/admin');
    console.log('Login com nova senha: SUCESSO');

  } catch (e) {
    console.error('Falha no teste:', e);
    process.exit(1);
  } finally {
    await browser.close();
  }
})();

File: tests/e2e/auth_flow.py
import asyncio
import os
from playwright.async_api import async_playwright

async def run_test():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        context = await browser.new_context(viewport={"width": 1280, "height": 1800})
        page = await context.new_page()
        
        print("--- Iniciando Teste E2E: Redefinição de Senha ---")
        try:
            await page.goto("http://localhost:8080/admin-login")
            print("Acessou /admin-login")

            await page.click('button:has-text("Esqueci minha senha admin")')
            print("Clicou em 'Esqueci minha senha'")

            await page.fill('input[placeholder="usuário"]', "admin")
            await page.click('button:has-text("Verificar Usuário")')
            print("Usuário 'admin' verificado")

            await page.fill('input[placeholder="mínimo 6 caracteres"]', "nova_senha_2026")
            await page.click('button:has-text("Salvar Nova Senha")')
            print("Senha alterada com sucesso")

            await page.click('button:has-text("Voltar ao login admin")')
            await page.fill('input[placeholder="usuário"]', "admin")
            await page.fill('input[placeholder="••••••••"]', "nova_senha_2026")
            await page.click('button[type="submit"]')
            
            await page.wait_for_url("**/admin", timeout=10000)
            print("Login com nova senha: SUCESSO")
        except Exception as e:
            print(f"Falha no teste: {e}")
            exit(1)
        finally:
            await browser.close()

if __name__ == "__main__":
    asyncio.run(run_test())
